src/jobsautoreport/report.py

```diff
@@ -1,31 +1,215 @@
-from datetime import datetime, timedelta
-from typing import Optional
+from datetime import datetime
+from typing import Any
 
+from pydantic import BaseModel
+
+from jobsautoreport.models import JobState, JobType
 from jobsautoreport.query import Querier
+from prowjobsscraper.event import JobDetails
+
+
+class Report(BaseModel):
+
+    number_of_e2e_or_subsystem_periodic_jobs_triggered: int
+    number_of_successful_e2e_or_subsystem_periodic_jobs_triggered: int
+    number_of_failures_e2e_or_subsystem_periodic_jobs_triggered: int
+    success_rate_for_e2e_or_subsystem_periodic_jobs: float
+    top_10_failed_e2e_or_subsystem_periodic_jobs: list[tuple[str, float]]
+    number_of_e2e_or_subsystem_presubmit_jobs_triggered: int
+    number_of_successful_e2e_or_subsystem_presubmit_jobs_triggered: int
+    number_of_failures_e2e_or_subsystem_pre_submit_jobs_triggered: int
+    number_of_rehearsal_jobs_triggered: int
+    success_rate_for_e2e_or_subsystem_presubmit_jobs: float
+    top_10_failed_e2e_or_subsystem_presubmit_jobs: list[tuple[str, float]]
+    top_5_most_triggered_e2e_or_subsystem_jobs: list[tuple[str, int]]
```

Comment:
at least to my eyes it feels redundant saving each metric in a field
can't the report be something like this?
```
def get_report(...):
return f"""
    Full report:
        Rehearse jobs count: {self.get_number_of_rehearsal_jobs_triggered()}.
        e2e/subsystem jobs count: {self.get_number_of_rehearsal_jobs_triggered()}.
        ...
"""
```

Reply:
We have some advantages by returning an object
https://github.com/openshift-assisted/prow-jobs-scraper/pull/55#discussion_r1046127175

Reply:
I agree it's an advantage over returning a dictionary
but imho there's an advantage in something much more simple which is constructing the report and sending it right away. If several components were to use those members then it would make more sense to store them (but it's not the case here)

Reply:
The idea is to decouple the aggregated data from the output to Slack. We could pass the Report to the Slack object directly without any intermediate object.

Reply:
ack. then let's keep it as-is